### ISSUE TITLE
Bug: Adding SSH key to ssh-agent on JupyterHub doesn't persist across terminal sessions

### ISSUE THREAD

**Issue by lauriemerrell** [cal-itp/data-infra#903]
**Describe the bug**
When following the setup instructions for saving code from JupyterHub ([Cal-ITP docs](https://docs.calitp.org/data-infra/analytics_tools/saving_code.html), [GitHub docs](https://docs.calitp.org/data-infra/analytics_tools/saving_code.html)), adding the SSH key to the ssh-agent doesn't seem to persist across terminal sessions. If I add the key in one terminal and then open a new terminal, the new window will continue to ask me for the key passphrase every time I perform a git action. 

**To Reproduce**
1. Go to JupyterHub
2. Open a terminal
3. Start ssh-agent: `eval "$(ssh-agent -s)"`
4. Add the key to the agent: `ssh-add ~/.ssh/id_ed25519`
5. Will be prompted for key's passphrase if configured; enter passphrase
6. Open a new terminal window in the same JupyterHub session
7. Test connection: `ssh -T git@github.com`
8. Will be prompted for passphrase again

**Expected behavior**
After adding the SSH key to the ssh-agent once, user should no longer be prompted for key passphrase in any subsequent terminal or JupyterHub session.

**Additional context**
When following the GitHub setup instructions initially, I ran into this error that they mention and followed their recommended steps to address:

> 
> Note: If you see an error like this
> 
> /Users/USER/.ssh/config: line 16: Bad configuration option: usekeychain
> 
> add an additional config line to your Host * section:
> 
> Host *
>   IgnoreUnknown UseKeychain
> 

 I assume that's related.

**Comment by machow** [comment]
I think this might be in line with how `ssh-agent` works (afaik it only stores your info for a single session), but I'm pretty unfamiliar. Broadly speaking, there might be another tool we could install to support storing passwords across sessions? (e.g. the linux tool keychain; not to confuse w/ the mac keychain).

afaik, UseKeychain is a mac specific option

**Comment by lauriemerrell** [comment]
Yeah, agree with Michael -- the GitHub docs for the Linux setup don't actually have you create the .ssh/config file at all. So my comment above about the UseKeychain argument is not actually relevant.
